packages/functoria/functoria.4.2.1/opam

```diff
@@ -0,0 +1,57 @@
+opam-version: "2.0"
+maintainer:   "Gabriel Radanne <drupyog@zoho.com>"
+authors:      [ "Thomas Gazagnaire"
+                "Anil Madhavapeddy"
+                "Dave Scott"
+                "Thomas Leonard"
+                "Gabriel Radanne" ]
+homepage:     "https://github.com/mirage/mirage"
+bug-reports:  "https://github.com/mirage/mirage/issues"
+dev-repo:     "git+https://github.com/mirage/mirage.git"
+doc:          "https://mirage.github.io/mirage/"
+license:      "ISC"
+tags:         ["org:mirage"]
+available: opam-version >= "2.1.0"
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+  ["env" "INSIDE_FUNCTORIA_TESTS=1" "dune" "exec" "-p" name "-j" jobs "--"
+     "test/functoria/e2e/test.exe"] {with-test}
+]
+
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.9.0" | (with-test & >= "3.0.0")}
```

Comment:
I don't think `| with-test & other-constraint` is what you want. If you substitute you get:
* "dune" {>= "2.9.0" | false} when with-test is false
* "dune" {>= "2.9.0" | >= "3.0.0") when with-test is true, which simplifies into `"dune" {>= "2.9.0"}`
```suggestion
  "dune" {>= "2.9.0"}
  "dune" {with-test & >= "3.0.0"}
```